src/openms/include/OpenMS/CHEMISTRY/ModificationsDB.h

```diff
@@ -80,6 +80,16 @@ namespace OpenMS
       return db_;
     }
 
+    inline static ModificationsDB * getCustomInstance(OpenMS::String unimod_file, OpenMS::String psimod_file)
```

Comment:
I think there are several options that make sense:
- use the static version at every place in OpenMS as it is done now. If one wants to use a custom ModificationsDB, then he can either modify the xml we provide, or we provide a mechanism (environment variable or similar like used for the OpenMS tmp) that redirects the loading of the DB to a custom one.
or:
- use the static version but provide a method to add modifications from a custom DB

Hmm I took a closer look and the issues don't seem to be that minor as I first thought.
If we want to drop static, then I guess we need to go the whole way of passing the ptr to classes - which is indeed awful and nearly impossible without changing everything. While I really dislike the way it is done right now I think it is best to stick with one of the options above.
If only modifications need to be added I actually would prefer the second option (while the first would give the guarantee that all code uses the same db).